packages/framework/pr_tools/trilinosprhelpers/TrilinosPRConfigurationStandard.py

```diff
@@ -24,65 +24,77 @@ def execute_test(self):
-        cmd = ['ctest', '-S', 'simple_testing.cmake',
-                      '-Dbuild_name={}'.format(self.pullrequest_build_name),
-                      '-Dskip_by_parts_submit=OFF',
-                      '-Dskip_update_step=ON',
-                      '-Ddashboard_model=Experimental',
-                      '-Ddashboard_track={}'.format(self.arg_pullrequest_cdash_track),
-                      '-DPARALLEL_LEVEL={}'.format(self.concurrency_build),
-                      '-DTEST_PARALLEL_LEVEL={}'.format(self.concurrency_test),
-                      '-Dbuild_dir={}/pull_request_test'.format(self.arg_workspace_dir),
-                      '-Dconfigure_script=' +
-                          os.path.join(self.arg_workspace_dir,
-                                       self.config_script),
-                      '-Dpackage_enables=' + self.arg_filename_packageenables,
-                      '-Dsubprojects_file=' + self.arg_filename_subprojects
-                    ]
-
-        print("--- ctest command:")
-        print("--- cmd = {}".format(" \\\n   ".join(cmd)))
-        print("")
+        cmd = ['ctest',
+                "-S", f"{self.arg_ctest_driver}",
+               f"-Dsource_dir:PATH={self.arg_source_dir}",
+               f"-Dbuild_dir:PATH={self.arg_build_dir}",
+               f"-Dbuild_name:STRING={self.pullrequest_build_name}",
+                "-Dskip_by_parts_submit:BOOL=OFF",
+                "-Dskip_update_step:BOOL=ON",
+                "-Ddashboard_model:STRING='Experimental'",
+               f"-Ddashboard_track:STRING='{self.arg_pullrequest_cdash_track}'",
+               f"-DPARALLEL_LEVEL:STRING={self.concurrency_build}",
+               f"-DTEST_PARALLEL_LEVEL:STRING={self.concurrency_test}",
+                "-Dconfigure_script:FILEPATH=" + os.path.join(self.arg_workspace_dir, self.config_script),
+               f"-Dpackage_enables:FILEPATH={self.arg_filename_packageenables}",
+               f"-Dsubprojects_file:FILEPATH={self.arg_filename_subprojects}"
+             ]
+
+        self.message( "--- ctest version:")
+        if not self.args.dry_run:
+            try:
+                # Check the version of CTest
+                subprocess.check_call(['ctest', '--version'], env=os.environ)
+            except subprocess.CalledProcessError as err:
+                self.message("-- ERROR: `ctest --version` failed.")
+                self.message(err)
+                return 1
+
+        self.message( "--- ctest command:")
+        self.message(f"--- pwd:\n{os.getcwd()}")
+        self.message( "--- cmd:\n{}".format(" \\\n   ".join(cmd)))
 
         if not self.args.dry_run:
             try:
                 subprocess.check_call(cmd, env=os.environ)
                 # Note: check_call will throw an exception if there's a problem.
-            except:
-                print("--- ctest command failed!")
+                self.message("--- OK")
+            except subprocess.CalledProcessError as err:
```

Comment:
Could this be missing an exception and causing the autotester to report the tests as passing?

Reply:
@e10harvey Nah, that shouldn't be the problem... The rest of that `except` block will cause `execute_test()` to return a 1:

https://github.com/william76/Trilinos/blob/37d63da3e7be4531303ea33189d20b23deb65d13/packages/framework/pr_tools/trilinosprhelpers/TrilinosPRConfigurationStandard.py#L73-L79

And back up in `PullRequestLinuxDriverTest.py` the status of `execute_test()` is captured and gets passed up and that's what should be getting returned when the script exits.

https://github.com/william76/Trilinos/blob/37d63da3e7be4531303ea33189d20b23deb65d13/packages/framework/pr_tools/PullRequestLinuxDriverTest.py#L288-L301

It looks like the output of this on CDash indicates that the Pytest run should be exiting with a nonzero exit code:
https://testing.sandia.gov/cdash/test/92263035

```
--- STATUS: 1

--------------------------------------------------------------------------------

TEST_0: Return code = 1
TEST_0: Pass criteria = Match REGEX {=+ [0-9]+ passed in} [FAILED]
TEST_0: Result = FAILED

================================================================================

OVERALL FINAL RESULT: TEST FAILED (TrilinosFrameworkTests_PullRequestLinuxDriver_UnitTests)

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

And the Jenkins job's console output ( https://ascic-jenkins.sandia.gov/job/trilinos-folder/job/python-3/196/console ) has this in the 'critical' part: 

```
00:00:50.059 +----------------------------------------------------------+
00:00:50.059 + START test step
00:00:50.059 +----------------------------------------------------------+
00:00:59.130 +----------------------------------------------------------+
00:00:59.130 + END test step - SUCCESS
00:00:59.130 +----------------------------------------------------------+
```

But I'm not sure how the test is getting launched in the new configuration -- IIRC the _old_ framework submitted the Python tests through something in `commontools/framework`... I'm not sure if @prwolfe has changed how this was launched but I saw this bit:

https://github.com/william76/Trilinos/blob/37d63da3e7be4531303ea33189d20b23deb65d13/commonTools/framework/CMakeLists.txt#L72-L83

Is this what we're running now for the Python tests?  If so, I wonder if we should uncomment the `#ALWAYS_FAIL_ON_NONZERO_RETURN` entry?

@prwolfe if you can send me some details on how that python test is getting launched it'd be useful. The new SimpleTestingRefactor should be doing all the same things the old one is.  The testing definitely shows the failure on CDash but that error somehow isn't getting propagated up to the Jenkins job.